### PR TITLE
ft: Add fullscreen screenshot capture

### PR DESCRIPTION
**Pull request by alwayswannaly**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -25,7 +25,7 @@ <h2>Captured Screenshot:</h2>
 
   <!-- Include the Webpack bundle -->
    <!-- Using the latest CDN  -->
-  <script src="https://cdn.jsdelivr.net/npm/exam-guard@4.0.0/dist/proctor.js"></script>
+   <script src="proctor.js"></script>
```
Comment (alwayswannaly) [inline public/index.html:28]: TODO: undo this change

File: src/utils/compatibilityModal.js [hunk 2]
```diff
@@ -4,16 +4,39 @@ import failureMark from '../assets/images/failMark.svg';
...
+  } else {
+    screenshareHTML = /* html */`
+      <div class="compatibility-modal__screenshare-info">
+        <div>Test your full screen share setup by clicking here&nbsp;<button id="fullscreen-share-button"> Share Full Screen </button></div>
+        <ul>
+          <li>You need to provide your full screen access</li>
+          <li>You'll be asked to share your full screen once the test starts</li>
+        </ul>
+      </div>
+    `;
```
Comment (aswanth9495) [inline src/utils/compatibilityModal.js:23]: Use templates here

File: src/utils/compatibilityModal.js [hunk 3]
```diff
@@ -32,14 +58,19 @@ export function setIconsForChecks(passedChecks = {}, checks = {}) {
...
+    fullscreenShareIcon.src = 'display: none';
+  }
 }
 
 export function setupCompatibilityCheckModal(onContinueClick, config = {}) {
   if (!config.enable) {
     return;
   }
+  const { mockModeEnabled } = config;
+  console.log(config);
```
Comment (aswanth9495) [inline src/utils/compatibilityModal.js:71]: Console

File: src/utils/screenshotV2.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+  SCREEN_SHARE_STREAM_ENDED: 'SCREEN_SHARE_STREAM_ENDED',
+  SCREEN_SHARE_FAILED: 'SCREEN_SHARE_FAILED',
+  SCREEN_SHARE_DENIED: 'SCREEN_SHARE_DENIED',
+  ROLLING_WINDOW_STRATEGY_LENGTH_EXCEEDED: 'ROLLING_WINDOW_STRATEGY_LENGTH_EXCEEDED',
+};
+
+const STRATEGIES = {
+  RETRY_STRATEGY: 'RETRY_STRATEGY',
+  ROLLING_WINDOW_STRATEGY: 'ROLLING_WINDOW_STRATEGY',
+};
```
Comment (aswanth9495) [inline src/utils/screenshotV2.js:14]: Add to constants file

File: src/index.js [hunk 5]
```diff
@@ -240,6 +245,8 @@ export default class Proctor {
       enabled: true,
       frequency: SNAPSHOT_SCREENSHOT_FREQUENCY,
       resizeTo: DEFAULT_SCREENSHOT_RESIZE_OPTIONS,
+      screenshareClickHandler: () => {},
+      screenshareRequestHandler: async () => {},
```
Comment (aswanth9495) [inline src/index.js:249]: We can move them as class methods
